### PR TITLE
Add compliant and noncompliant examples of javascript/insecure-temp-file@v1.0

### PR DESCRIPTION
**Pull request by Kamlesh-Gour**
*Description of changes:*
* Rules Added: insecure-temp-file@v1.0
* Added new functions for compliant and non compliant examples
* Added .gitignore

### CODE REVIEW

File: src/detectors/insecure_temporary_file/insecure_temporary_file.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+       // ...
+    })
+}
+// {/fact}
+
+// {fact rule=insecure-temp-file@v1.0 defects=0}
+var fs = require('fs')
+var tmp = require('tmp')
+function insecureTempFileCompliant() {
+    // Compliant: tmp library to securely create or read a temporary files.
```
Comment (ctlai95) [inline src/detectors/insecure_temporary_file/insecure_temporary_file.js:21]:
```suggestion
    // Compliant: tmp library to securely create or read temporary files.
```

File: src/detectors/insecure_temporary_file/insecure_temporary_file.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// {fact rule=insecure-temp-file@v1.0 defects=1}
+var fs = require('fs')
+function insecureTempFileNoncompliant() {
+    // Noncompliant: the global directory path is given for opening a file or creating a file the case can be vulnerable to injection attacks.
```
Comment (ctlai95) [inline src/detectors/insecure_temporary_file/insecure_temporary_file.js:9]:
```suggestion
    // Noncompliant: the global directory path is given for opening a file or creating a file which can be vulnerable to injection attacks.
```

**Review by ctlai95 — CHANGES_REQUESTED** [review]

File: src/detectors/insecure_temporary_file/insecure_temporary_file.js (hunk 2 above)
Comment (Kamlesh-Gour) [reply src/detectors/insecure_temporary_file/insecure_temporary_file.js]: Updated

File: src/detectors/insecure_temporary_file/insecure_temporary_file.js (hunk 1 above)
Comment (Kamlesh-Gour) [reply src/detectors/insecure_temporary_file/insecure_temporary_file.js]: Updated

**Review by mj632 — APPROVED** [review]

**Pull request merged**